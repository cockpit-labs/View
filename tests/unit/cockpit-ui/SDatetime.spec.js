import { shallowMount } from '@vue/test-utils'
import SDatetime from '@/plugins/cockpit-ui/components/datetime/SDatetime'

describe('Datetime', () => {
  const wrapper = shallowMount(SDatetime, {
    propsData: {
      value: '2020-01-01',
      placeholder: 'Set a date'
    },
    mocks: {
      $t: key => key
    }
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain a local value equal to value prop', () => {
    expect(wrapper.vm.localValue).toBe(wrapper.props().value)
  })

  it('should emit an input event after an update of the value', () => {
    const spy = jest.fn()
    wrapper.vm.$on('input', spy)
    wrapper.setData({ localValue: '2020-12-31' })
    wrapper.vm.commitValue()
    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('can be reset', () => {
    const spy = jest.fn()
    wrapper.vm.$on('input', spy)
    wrapper.vm.reset()
    expect(wrapper.vm.localValue).toBeNull()
    expect(spy).toHaveBeenCalledTimes(1)
  })
})
