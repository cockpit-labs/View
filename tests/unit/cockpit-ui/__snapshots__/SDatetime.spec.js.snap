// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datetime should match snapshot 1`] = `
<div
  class="s-datetime"
>
  <datetime-stub
    auto="true"
    backdropclick="true"
    hourstep="1"
    inputclass=""
    inputstyle=""
    minutestep="1"
    phrases="[object Object]"
    placeholder="Set a date"
    type="date"
    value="2020-01-01"
    valuezone="UTC"
    weekstart="1"
    zone="local"
  />
</div>
`;
